packages/framework/src/Model/Product/Recalculation/ProductRecalculationRepository.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product\Recalculation;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Shopsys\FrameworkBundle\Model\Product\Product;
+
+class ProductRecalculationRepository
+{
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $em
+     */
+    public function __construct(
+        protected readonly EntityManagerInterface $em,
+    ) {
+    }
+
+    /**
+     * @param int[] $productIds
+     * @return int[]
+     */
+    public function getIdsToRecalculate(array $productIds): array
```

Comment:
It would be nice to have a test for this method. Or at least an issue for creating the test :smile: